rfcs/20240326-block-level-api/README.md

```diff
@@ -0,0 +1,436 @@
+# Proposal for block level APIs
+
+## Overview
+
+In order to implement DNN operations, it is generally useful to rely
+on a set of highly optimized basic building blocks that are then
+composed together. Such a set is defined and discussed in [^1][^2],
+and is currently used in oneDNN CPU backend to implement Matmul,
+Convolution, Inner product and RNN primitives.  In particular, the
+main operation batch-reduce GEneral Matrix Multiply operation (aka
+brgemm), is a very flexible operation that can be used in a variety of
+DNN operators. It is defined as folllow:
+
+```math
+D = \beta C + \alpha \sum_i A_i \cdot B_i + bias
```

Comment:
We need three buffers or four buffers here? I thought we only need three?
```suggestion
C = \beta C + \alpha \sum_i A_i \cdot B_i + bias
```

Reply:
The need for D buffer stems from post-ops and output conversion fusion. To simplify, I will drop the D in that initial formula.